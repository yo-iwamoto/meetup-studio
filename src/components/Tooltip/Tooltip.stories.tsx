import { Tooltip, TooltipProvider } from './Tooltip';
import { userEvent } from '@storybook/testing-library';
import type { Meta, StoryObj } from '@storybook/react';

const meta = {
  component: Tooltip,
  decorators: [(story) => <TooltipProvider>{story()}</TooltipProvider>],
} satisfies Meta<typeof Tooltip>;

export default meta;

export const Default = {
  args: {
    label: 'content',
    children: (
      <button type='button' className='rounded-lg bg-orange-500 p-2 text-white'>
        Trigger
      </button>
    ),
  },
} satisfies StoryObj<typeof meta>;

export const Focused = {
  ...Default,
  play: async () => {
    await userEvent.tab();
  },
} satisfies StoryObj<typeof meta>;
