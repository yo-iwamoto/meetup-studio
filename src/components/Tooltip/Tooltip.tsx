import * as RadixTooltip from '@radix-ui/react-tooltip';
import type { ComponentProps, PropsWithChildren, ReactNode } from 'react';

type Props = PropsWithChildren<{
  label: ReactNode;
  arrow?: boolean;
}>;

export const Tooltip = ({ label, arrow = false, children }: Props) => {
  return (
    <RadixTooltip.Root delayDuration={0}>
      <RadixTooltip.Trigger asChild>{children}</RadixTooltip.Trigger>

      <RadixTooltip.Portal>
        <RadixTooltip.Content className='mb-3 rounded-md bg-black bg-opacity-80 px-2 py-1.5 text-white'>
          {label}
          {arrow && <RadixTooltip.Arrow />}
        </RadixTooltip.Content>
      </RadixTooltip.Portal>
    </RadixTooltip.Root>
  );
};

export const TooltipProvider = (props: ComponentProps<typeof RadixTooltip.Provider>) => {
  return <RadixTooltip.Provider {...props} />;
};
