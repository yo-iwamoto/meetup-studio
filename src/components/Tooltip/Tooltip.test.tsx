import * as stories from './Tooltip.stories';
import { act, render } from '@testing-library/react';
import { composeStories } from '@storybook/react';

const { Default, Focused } = composeStories(stories);

describe('Tooltip', () => {
  it('renders unchanged', () => {
    const { queryByRole, asFragment } = render(<Default />);

    expect(queryByRole('tooltip')).toBeNull();
    expect(asFragment()).toMatchSnapshot();
  });

  it('tooltip should be visible when focused', async () => {
    const { getByRole, container } = render(<Focused />);

    await act(() => Focused.play({ canvasElement: container }));

    expect(getByRole('tooltip', { name: 'content' })).toBeDefined();
  });
});
