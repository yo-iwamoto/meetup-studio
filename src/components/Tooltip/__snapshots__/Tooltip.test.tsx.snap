// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tooltip > renders unchanged 1`] = `
<DocumentFragment>
  <button
    class="rounded-lg bg-orange-500 p-2 text-white"
    data-state="closed"
    type="button"
  >
    Trigger
  </button>
</DocumentFragment>
`;
